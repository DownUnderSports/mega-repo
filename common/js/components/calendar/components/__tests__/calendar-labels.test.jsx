import React from 'react'
import dateFns from 'date-fns'
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils'
import { CalendarLabels } from '../'

describe('common/js/components - Calendar::Labels', () => {
  const div = document.createElement('div');

  const createLabels = ({...props}) => {
    return renderer.create(
      <CalendarLabels {...props} />
    )
  }

  it('renders a calendar-labels div', () => {
    const rendered = createLabels()
    expect(rendered).toBeTruthy()
    expect(rendered.toJSON().type).toBe("div")
    expect(rendered.toJSON().props.className.split(' ')[0]).toEqual("calendar-labels")
  })

  it('is a calendar row', () => {
    const rendered = createLabels(),
          classes = rendered.toJSON().props.className.split(' ')
    let rowed = false
    for(let i = 0; i < classes.length; i++) {
      if(classes[i] === 'calendar-row') rowed = true
    }
    expect(rowed).toBeTruthy()
  })

  it('has exactly 7 labels', () => {
    const children = createLabels().root.findAllByProps({
      'data-purpose': 'calendar-label'
    });
    expect(children.length).toEqual(7)
    for(let i = 0; i < 7; i++) {
      expect(children[i].props.className).toEqual('calendar-col col-center')
    }
  })

  it('takes a startDate', () => {
    const d = new Date(),
          child = createLabels({startDate: d}).root.findAllByProps({
            'data-purpose': 'calendar-label'
          })[0].findByProps({
            className: 'col-overflow'
          });

    expect(child.children[0]).toEqual(dateFns.format(d))
  })

  it('takes a labelFormat', () => {
    const startDate = new Date(),
          labelFormat = 'd',
          child = createLabels({startDate, labelFormat}).root.findAllByProps({
            'data-purpose': 'calendar-label'
          })[0].findByProps({
            className: 'col-overflow'
          });

    expect(child.children[0]).toEqual(dateFns.format(startDate, labelFormat))
  })

  it('is snapshotable', () => {
    const tree = createLabels()
    .toJSON();
    expect(tree).toMatchSnapshot()
  })
})
