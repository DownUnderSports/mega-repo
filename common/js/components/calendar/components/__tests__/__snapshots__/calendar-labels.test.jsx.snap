// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - Calendar::Labels is snapshotable 1`] = `
<div
  className="calendar-labels calendar-row"
>
  <div
    className="calendar-col col-center"
    data-purpose="calendar-label"
  >
    <div
      className="col-overflow"
    >
      Invalid Date
    </div>
  </div>
  <div
    className="calendar-col col-center"
    data-purpose="calendar-label"
  >
    <div
      className="col-overflow"
    >
      Invalid Date
    </div>
  </div>
  <div
    className="calendar-col col-center"
    data-purpose="calendar-label"
  >
    <div
      className="col-overflow"
    >
      Invalid Date
    </div>
  </div>
  <div
    className="calendar-col col-center"
    data-purpose="calendar-label"
  >
    <div
      className="col-overflow"
    >
      Invalid Date
    </div>
  </div>
  <div
    className="calendar-col col-center"
    data-purpose="calendar-label"
  >
    <div
      className="col-overflow"
    >
      Invalid Date
    </div>
  </div>
  <div
    className="calendar-col col-center"
    data-purpose="calendar-label"
  >
    <div
      className="col-overflow"
    >
      Invalid Date
    </div>
  </div>
  <div
    className="calendar-col col-center"
    data-purpose="calendar-label"
  >
    <div
      className="col-overflow"
    >
      Invalid Date
    </div>
  </div>
</div>
`;
